src/docks/timelinedock.cpp

```diff
@@ -1165,32 +1191,14 @@ void TimelineDock::insert(int trackIndex, int position, const QString &xml, bool
         properties.set(kShotcutSkipConvertProperty, 1);
         MAIN.open(Util::removeFileScheme(url), &properties, false /* play */ );
     }
+  
+    if (position < 0)
+        position = m_position;
+    if (m_model.trackList().size() == 0)
+        position = 0;
 
-    if (MLT.isSeekableClip() || MLT.savedProducer() || !xml.isEmpty()) {
-        QString xmlToUse;
-        QScopedPointer<TimelineSelectionBlocker> selectBlocker;
-        if (xml.isEmpty()) {
```

Comment:
I don't understand this. I am still able to drop to the timeline from both Source player and external file. Maybe I am missing something.

Reply:
Did you see everywhere this method is called? There are places where it is called from QML and passed an argument for `xml`. You cannot simply ignore it.

Reply:
The reason I had discarded it was because I removed the ` const QString &xml` parameter in the insertCommand but now I see what is wrong with that.
I could create a function `QVector<Mlt::Producer>convertUrlsToProducers()` that will use that xml argument.